README.md

```diff
@@ -1,5 +1,90 @@
-# browserstack-bitrise-xcui-step
+# BrowserStack App Automate - XCUI
 
-The official Browserstack xcui Bitrise Step, using which users can integrate their Bitrise CI/CD pipelines with Browserstack
+## 🧩 Get started
 
-# browserstack-bitrise-xcui-step
+Add this step directly to your workflow in the [Bitrise Workflow Editor](https://devcenter.bitrise.io/en/steps-and-workflows/introduction-to-workflows.html).
+
+OR
+
+You can also run this step directly with [Bitrise CLI]([https://github.com/bitrise-io/bitrise](https://github.com/bitrise-io/bitrise#bitrise-offline-cli)).
+
+## Run XCUI tests on BrowserStack
+
+<details>
+<summary>Description</summary>
+
+Run your XCUI tests on BrowserStack App Automate. This step collects the built IPA from `$BITRISE_IPA_PATH` and the output bundle file from `$BITRISE_TEST_BUNDLE_PATH` environment variables.
+
+## Configure the Step
+
+Before configuring this step, make sure you install [Bitrise CLI](https://github.com/bitrise-io/bitrise) on your machine.
+
+Complete the following steps:
+
+1. Open the Workflow you want to use in the Workflow Editor.
+​
+2. Add the [Xcode Archive & Export for iOS](https://www.bitrise.io/integrations/steps/xcode-archive) and [Xcode Build for testing for iOS](https://www.bitrise.io/integrations/steps/xcode-build-for-test) steps to your workflow and configure them.
+​
+3. Add the **BrowserStack App Automate - XCUI** step below the **Xcode Archive & Export for iOS** and **Xcode Build for testing for iOS** steps.
+​
+4. Add your BrowserStack Username and Access Key in the **Authentication** step input.
+​
+5. For **App & Test Suite** step inputs, the **Xcode Archive & Export for iOS** and **Xcode Build for testing for iOS** steps export an APK and a test APK, respectively. Thus, their paths get automatically set in the APK path and Test APK path input fields. <br /><br />If you are not using  **Xcode Archive & Export for iOS** and **Xcode Build for testing for iOS** steps, make sure the **App path** input points to the path of your app and test suite files. In the case of the runner app, it should be in `<any_path>/Debug-iphoneos` directory if the user is providing an absolute path.
```

Comment:
For iOS app input, the BITRISE_IPA_PATH output from  **Xcode Archive & Export for iOS** step exports the ipa and for XCUI test suite input, the BITRISE_TEST_BUNDLE_PATH output from **Xcode Build for testing for iOS** step exports the test-suite.
Thus, their paths get automatically set in the iOS app and XCUI test suite input fields. <br /><br />If you are not using  **Xcode Archive & Export for iOS** and **Xcode Build for testing for iOS** steps, make sure the **iOS app** input points to the path of your app (.ipa file) and **XCUI test suite** input points to the test suite runner file. In the case of the runner app, it should be in `<any_path>/Debug-iphoneos` directory if the user is providing an absolute path.